src/dmd/expressionsem.d

```diff
@@ -11694,6 +11694,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         case Terror:
             return setError();
 
+        case Tarray, Tsarray:
+            result = exp.incompatibleTypes();
+            exp.errorSupplemental("`in` is only allowed on associative arrays");
+            exp.errorSupplemental("perhaps use `find` or `canFind` from `std.algorithm`");
```

Comment:
Possible suggestion, would it be more clearer to give a full example, such as:
```
"perhaps use `std.algorithm.find(%s, %s)` instead", exp.e1.toChars(), exp.e2.toChars()
```

Reply:
Good suggestion